src/main/java/org/generationcp/middleware/manager/WorkbenchDataManagerImpl.java

```diff
@@ -1153,4 +1166,43 @@ public WorkbenchSidebarCategoryLink getWorkbenchSidebarLinksById(final Integer w
 	public void saveOrUpdateUserRole(final UserRole userRole) {
 		this.getUserRoleDao().saveOrUpdate(userRole);
 	}
+
+	@Override
+	public Permission getPermission(final Integer permissionId) {
+		return this.getPermissionDao().getById(permissionId);
+	}
+
+	@Override
+	public List<Permission> getPermissions() {
+		return this.getPermissionDao().getAll();
+	}
+
+	@Override
+	public Role saveOrUpdateRole(final Role role) {
```

Comment:
@vmaletta in particular I always prefer to have both save and update in 2 functions